### PR TITLE
Re-moving walls as edit_room doesn't work.

### LINKED ISSUE #2204: [Bug] Earthquake freezes the game
### Describe the issue

Gameplay freezes during an earthquake. This is not the first earthquake I experienced on this map. The other ones went away without any issue.

The game is frozen (no one is moving) but the screen shakes when I move the mouse and the top menu bar is responsive.

```
An error has occurred!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
/home/jsimek/Development/CorsixTH/CorsixTH/Lua/room.lua:863: attempt to call a nil value (method 'getWallSetFromBlockId')
stack traceback:
        /home/jsimek/Development/CorsixTH/CorsixTH/Lua/room.lua:863: in method 'crashRoom'
        ...k/Development/CorsixTH/CorsixTH/Lua/entities/machine.lua:136: in function <...k/Development/CorsixTH/CorsixTH/Lua/entities/machine.lua:97>
        (...tail calls...)
        /home/jsimek/Development/CorsixTH/CorsixTH/Lua/world.lua:617: in method 'tickEarthquake'
        /home/jsimek/Development/CorsixTH/CorsixTH/Lua/world.lua:972: in method 'onTick'
        /home/jsimek/Development/CorsixTH/CorsixTH/Lua/app.lua:1152: in function </home/jsimek/Development/CorsixTH/CorsixTH/Lua/app.lua:1149>
        (...tail calls...)
        /home/jsimek/Development/CorsixTH/CorsixTH/Lua/app.lua:1052: in function </home/jsimek/Development/CorsixTH/CorsixTH/Lua/app.lua:1047>

Warning: No event handler for timer
```

### Steps to Reproduce

1. *Load save game attached to this ticket*

2. Earthquake is already in progress so you should experience the issue immediately.

### Expected Behaviour

Continue to play.

### Save Game

Save game attached to report: Yes

[earthquake.zip](https://github.com/CorsixTH/CorsixTH/files/9093085/earthquake.zip)

### CorsixTH Version

746a8ea1378c5c5bcd84b915ebe023f1cff6f58c

### Operating System

Arch Linux

### Theme Hospital Version

GOG.com

### Gamelog.txt

```bash
Compiled with api_version: 2682, arch: x86_64, audio: true, os: unix
SDL renderer: opengl
Lua 5.4 run with api version: 2682, game version: Trunk, savegame version: 171
Created game with savegame version 171.
Error in callback handler: 
/home/jsimek/Development/CorsixTH/CorsixTH/Lua/audio.lua:586: Could not load music file 'Sound/Midi/NIGHTSH.XMI' (Couldn't open /etc/timidity++/freepats.cfg)
stack traceback:
	[C]: in function 'coroutine.yield'
	/home/jsimek/Development/CorsixTH/CorsixTH/Lua/app.lua:1052: in function </home/jsimek/Development/CorsixTH/CorsixTH/Lua/app.lua:1047>
Warning: Trying to remove nonexistent room build callback (function: 0x558edc2dc230) from humanoid (table: 0x558edcc7cf00).
Warning: Trying to remove nonexistent room remove callback (function: 0x558edd8093b0) from humanoid (table: 0x558edcc7cf00).
Warning: Trying to remove nonexistent staff callback (function: 0x558edcf547c0) from humanoid (table: 0x558edcc7cf00).
Warning: Trying to remove nonexistent room build callback (function: 0x558ee0082c20) from humanoid (table: 0x558edec23c80).
Warning: Trying to remove nonexistent room remove callback (function: 0x558edb634e20) from humanoid (table: 0x558edec23c80).
Warning: Trying to remove nonexistent staff callback (function: 0x558edc5233e0) from humanoid (table: 0x558edec23c80).
Warning: Trying to remove nonexistent room build callback (function: 0x558edc70cb20) from humanoid (table: 0x558edcaa3140).
Warning: Trying to remove nonexistent room remove callback (function: 0x558edceda1d0) from humanoid (table: 0x558edcaa3140).
Warning: Trying to remove nonexistent staff callback (function: 0x558edbaee300) from humanoid (table: 0x558edcaa3140).
Warning: Trying to remove nonexistent room build callback (function: 0x558edca3e970) from humanoid (table: 0x558edbbddb40).
Warning: Trying to remove nonexistent room remove callback (function: 0x558ee0264f10) from humanoid (table: 0x558edbbddb40).
Warning: Trying to remove nonexistent staff callback (function: 0x558ede550750) from humanoid (table: 0x558edbbddb40).
Warning: Trying to remove nonexistent room build callback (function: 0x558eddb9b4e0) from humanoid (table: 0x558edbeab230).
Warning: Trying to remove nonexistent room remove callback (function: 0x558eddcab9c0) from humanoid (table: 0x558edbeab230).
Warning: Trying to remove nonexistent staff callback (function: 0x558edbeab9b0) from humanoid (table: 0x558edbeab230).
Warning: Trying to remove nonexistent room build callback (function: 0x558edbb92690) from humanoid (table: 0x558edbc4ce50).
Warning: Trying to remove nonexistent room remove callback (function: 0x558edbbddd90) from humanoid (table: 0x558edbc4ce50).
Warning: Trying to remove nonexistent staff callback (function: 0x558edf13b8b0) from humanoid (table: 0x558edbc4ce50).
Warning: Trying to remove nonexistent room build callback (function: 0x558ee05818a0) from humanoid (table: 0x558edbb1f800).
Warning: Trying to remove nonexistent room remove callback (function: 0x558edbbcae70) from humanoid (table: 0x558edbb1f800).
Warning: Trying to remove nonexistent staff callback (function: 0x558edb7a8a80) from humanoid (table: 0x558edbb1f800).
Warning: Trying to remove nonexistent room build callback (function: 0x558ede9b58e0) from humanoid (table: 0x558ede374f30).
Warning: Trying to remove nonexistent room remove callback (function: 0x558ee01d67a0) from humanoid (table: 0x558ede374f30).
Warning: Trying to remove nonexistent staff callback (function: 0x558edd8af240) from humanoid (table: 0x558ede374f30).
Warning: Trying to remove nonexistent room build callback (function: 0x558edbc04cd0) from humanoid (table: 0x558edb6c6ed0).
Warning: Trying to remove nonexistent room remove callback (function: 0x558edc608080) from humanoid (table: 0x558edb6c6ed0).
Warning: Trying to remove nonexistent staff callback (function: 0x558ede5bf790) from humanoid (table: 0x558edb6c6ed0).
Warning: Trying to remove nonexistent room build callback (function: 0x558ede3ac920) from humanoid (table: 0x558ee03e2900).
Warning: Trying to remove nonexistent room remove callback (function: 0x558edc500850) from humanoid (table: 0x558ee03e2900).
Warning: Trying to remove nonexistent staff callback (function: 0x558eddde2f40) from humanoid (table: 0x558ee03e2900).
Warning: Trying to remove nonexistent room build callback (function: 0x558edef191b0) from humanoid (table: 0x558edb935050).
Warning: Trying to remove nonexistent room remove callback (function: 0x558edbfca9b0) from humanoid (table: 0x558edb935050).
Warning: Trying to remove nonexistent staff callback (function: 0x558edcdef3e0) from humanoid (table: 0x558edb935050).
Warning: Trying to remove nonexistent room build callback (function: 0x558edc30fc00) from humanoid (table: 0x558ed5c1fd60).
Warning: Trying to remove nonexistent room remove callback (function: 0x558edb9594e0) from humanoid (table: 0x558ed5c1fd60).
Warning: Trying to remove nonexistent staff callback (function: 0x558edbdcf5f0) from humanoid (table: 0x558ed5c1fd60).
Warning: Trying to remove nonexistent room build callback (function: 0x558ee085ba00) from humanoid (table: 0x558eddb4c9a0).
Warning: Trying to remove nonexistent room remove callback (function: 0x558edea7f1d0) from humanoid (table: 0x558eddb4c9a0).
Warning: Trying to remove nonexistent staff callback (function: 0x558edd8d8c30) from humanoid (table: 0x558eddb4c9a0).
Savegame version is 171 (Trunk), originally it was 171 (Trunk).
Savegame version is 171 (Trunk), originally it was 171 (Trunk).
Savegame version is 171 (Trunk), originally it was 171 (Trunk).
Error in timer handler: 
/home/jsimek/Development/CorsixTH/CorsixTH/Lua/room.lua:863: attempt to call a nil value (method 'getWallSetFromBlockId')
stack traceback:
	/home/jsimek/Development/CorsixTH/CorsixTH/Lua/room.lua:863: in method 'crashRoom'
	...k/Development/CorsixTH/CorsixTH/Lua/entities/machine.lua:136: in function <...k/Development/CorsixTH/CorsixTH/Lua/entities/machine.lua:97>
	(...tail calls...)
	/home/jsimek/Development/CorsixTH/CorsixTH/Lua/world.lua:617: in method 'tickEarthquake'
	/home/jsimek/Development/CorsixTH/CorsixTH/Lua/world.lua:972: in method 'onTick'
	/home/jsimek/Development/CorsixTH/CorsixTH/Lua/app.lua:1152: in function </home/jsimek/Development/CorsixTH/CorsixTH/Lua/app.lua:1149>
	(...tail calls...)
	/home/jsimek/Development/CorsixTH/CorsixTH/Lua/app.lua:1052: in function </home/jsimek/Development/CorsixTH/CorsixTH/Lua/app.lua:1047>
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Alberth289346**
Fixes #2204

**Describe what the proposed change does**
- Move the wall types back to world, as edit_room isn't always available.
- Adds another savegame version
- Makes me sooo very tired of this stuff

### CODE REVIEW

**Comment by lewri** [comment]
https://github.com/CorsixTH/CorsixTH/blob/746a8ea1378c5c5bcd84b915ebe023f1cff6f58c/CorsixTH/Lua/room.lua#L863 -- had to track that back.

Probably the wall code should be somewhere else than edit room/world. Maybe there's another class it belongs in?
Couldn't walls be argued as being a piece of map.lua?

**Comment by Alberth289346** [comment]
Perhaps a separate class, there could be room for something "graphics" I guess

File: CorsixTH/Lua/world.lua [hunk 1]
```diff
@@ -2713,6 +2768,10 @@ function World:afterLoad(old, new)
     self.wall_set_by_block_id = nil
     self.wall_dir_by_block_id = nil
   end
+  if old < 172 then
+    self.wall_types = self.app.wall_types
+    self:initWallTypes()
+  end
```
Comment (lewri) [inline CorsixTH/Lua/world.lua:2774]: Arguably deleting the wall variables in one version then readding them in the next is not optimal or sensible now.
It goes against convention but can't we modify `if old < 171 then` code so this weirdness doesn't occur moving forward?

Else, if you're considering a new class or something I'm not sure what's the best course of action? Scuttle this PRs current code perhaps and bring in the new class as the 'fix'?

Comment (Alberth289346) [reply CorsixTH/Lua/world.lua:2774]: I am not considering anything currently, tbh I am quite done with fighting the language and its lack of adequate tools.

**Pull request merged**